indico/modules/attachments/templates/_attachments.html

```diff
@@ -8,10 +8,11 @@
 
 {% macro render_attachment(attachment, has_label=false, classes='') %}
     {% set previewable = (attachment.file.is_previewable and not g.static_site) or false %}
+    {% set visible_link = attachment.type.name == 'link' and (not attachment.is_self_protected or attachment.can_access(session.user)) %}
```

Comment:
```suggestion
    {% set visible_link = attachment.type.name == 'link' and (not attachment.is_protected or attachment.can_access(session.user)) %}
```

otherwise you reveal links in protected folders